### PR TITLE
Imx8m fixes

### PR DESCRIPTION
**Pull request by dbaluta**
Two fixes allowing us to progress on enabling SDMA.

First propagates some config fields to channel.\
Second enhances the state machine taking into consideration that channel 0 is special.

### CODE REVIEW

**Comment by dbaluta** [comment]
SOFCI TEST

File: src/drivers/imx/sdma.c [hunk 1]
```diff
@@ -478,11 +478,22 @@ static int sdma_start(struct dma_chan_data *channel)
 {
 	struct sdma_chan *pdata = dma_chan_get_data(channel);
 
-	if (channel->status == COMP_STATE_INIT)
-		return -EINVAL;
-
 	trace_sdma_error("sdma_start(%d)", channel->index);
 
+
```
Comment (ranj063) [inline src/drivers/imx/sdma.c]: @dbaluta nothing to do with this patch. But the trace above is not really an error is it?

File: src/drivers/imx/sdma.c [hunk 2]
```diff
@@ -508,8 +519,12 @@ static int sdma_stop(struct dma_chan_data *channel)
 {
 	struct sdma_chan *pdata = dma_chan_get_data(channel);
 
-	if (channel->status == COMP_STATE_INIT)
-		return -EINVAL;
+	/* do not try to stop multiple times */
+	if (channel->status != COMP_STATE_ACTIVE &&
```
Comment (ranj063) [inline src/drivers/imx/sdma.c]: Should this be || instead of &&?

File: src/drivers/imx/sdma.c (hunk 1 above)
Comment (dbaluta) [reply src/drivers/imx/sdma.c]: @ranj063 indeed it is not an error. It belongs to a separate patch.

File: src/drivers/imx/sdma.c (hunk 2 above)
Comment (dbaluta) [reply src/drivers/imx/sdma.c]: I think  && is correct. We arrive at sdma_stop from either COMP_STATE_ACTIVE or PAUSED. If there is any other state then this too, this means that stop was already called so we just exit.

Same thing happens for example in src/drivers/dw/dma.c

Comment (ranj063) [reply src/drivers/imx/sdma.c]: Ahh yes. I misread the code there. sorry for the noise!

**Comment by dbaluta** [comment]
SOFCI TEST

**Pull request merged**